modules/reitit-core/src/reitit/impl.cljc

```diff
@@ -163,18 +163,59 @@
   (->> m (remove (comp nil? second)) (into {})))
 
 ;;
-;; Path-parameters, see https://github.com/metosin/reitit/issues/75
+;; Parts (c) https://github.com/lambdaisland/uri/tree/master/src/lambdaisland/uri
 ;;
 
+#?(:clj
+   (def hex-digit
+     {0 "0" 1 "1" 2 "2" 3 "3"
+      4 "4" 5 "5" 6 "6" 7 "7"
+      8 "8" 9 "9" 10 "A" 11 "B"
+      12 "C" 13 "D" 14 "E" 15 "F"}))
+
+#?(:clj
+   (defn byte->percent [byte]
+     (let [byte (bit-and 0xFF byte)
+           low-nibble (bit-and 0xF byte)
+           high-nibble (bit-shift-right byte 4)]
+       (str "%" (hex-digit high-nibble) (hex-digit low-nibble)))))
```

Comment:
Meh, Github ate my comment here. Here it is again:

This looks like a lookup table... Why not go further and have a bigger lookup table?

```clj
(def +percents+ (into-array (map #(format "%%%02x" %) (range 255))))
(defn byte->percent [byte] (aget +percents+ byte))
```

Reply:
brilliant.

Reply:
except:

```
expected: (= "%E2%80%9C%D0%8C%CF%8D%D0%91%D0%87%E2%80%9D" (impl/url-encode "“ЌύБЇ”"))
  actual: java.lang.ArrayIndexOutOfBoundsException: -30
```

Reply:
Right, I forgot about Java's signed bytes.

```clj
(defn byte->percent [byte] (aget +percents+ (bit-and 0xFF byte)))
```

Reply:
```clj
expected: (= "%2F" (impl/url-encode "/"))
  actual: (not (= "%2F" "%ffffffffffffff30"))
```